src/services/mongoose/wallet-invoices.ts

```diff
@@ -49,5 +51,81 @@ export const WalletInvoicesRepository = (): IWalletInvoicesRepository => {
     }
   }
 
-  return { persist, findByPaymentHash }
+  async function* findPendingByWalletId(
+    walletId: WalletId,
+  ): AsyncGenerator<WalletInvoice> | RepositoryError {
+    let pending
+    try {
+      pending = InvoiceUser.find({ uid: walletId, paid: false }).cursor({
+        batchSize: 100,
+      })
+    } catch (error) {
+      return new RepositoryError(error)
```

Comment:
Errors that we catch and wrap always have the `Unknown` prefix. Here it should be `UnknownRepositoryError`.

Reply:
so this is a general problem, in current use cases you are only validating `RepositoryError` and... `UnknownRepositoryError` and `CouldNotFindError` do not extends from `RepositoryError`  so... what do you prefer? extend repo errors from RepositoryError or use isRepoError to do validations

Reply:
Don't use `isRepoError` use `xxx instanceof Error`.
`UnknownRepositoryError` and `CouldNotFindError` should extend from `RepositoryError` if they don't this is a mistake.

Still we should only differentiate which error was being returned if this is meaningful (like translating in the GQL layer or something) otherwise always use `instanceof Error`

Reply:
done